pkg/statscombiner/statscombiner.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statscombiner
```

Comment:
I was wondering if it could have a more generic name. Maybe datacombiner or at least not related to "stats". In the end, it has nothing to do with stats. It is just combining `any` kind of data, don't you think?

Reply:
Hmm, that's true. It does however have this time component (ttl, expecting "events" and periodic requests). Naming it too generic might be confusing as the use case is pretty specific.

Reply:
Similarly to the answer below: what about `snapshotcombiner`, then? It simply overwrites old snapshots as new get in and keeps them only for a certain amount of (combined) requests.